sklearn/model_selection/_validation.py

```diff
@@ -799,22 +801,23 @@ def _score(estimator, X_test, y_test, scorer, error_score="raise"):
                 )
 
     error_msg = "scoring must return a number, got %s (%s) instead. (scorer=%s)"
-    if isinstance(scores, dict):
-        for name, score in scores.items():
-            if hasattr(score, "item"):
+    if not isinstance(estimator, (_MultiOutputEstimator, _BaseChain)):
```

Comment:
I assume that we should leverage the tag "multioutput_only" and "multioutput" and the shape of the score because we should accepting 2 dimensional scores.

So I see three bugs here:

- the "multioutput" / "multioutput_only" should be True of those regressors and classifiers
- we should have a common test that check the if "multioutput_only" is True then, "multioutput" should be set to True as well
- `Pipeline` should expose the tag of the last step for the multioutput.